### PR TITLE
L5.1 and Lumen support

### PR DESCRIPTION
**Pull request by vluzrmos**

### CODE REVIEW

**Comment by vluzrmos** [comment]
I'll remove php 5.4 from travis.yml

File: src/TrustedProxyServiceProvider.php [hunk 1]
```diff
@@ -15,7 +15,10 @@ public function boot()
     {
         $source = realpath(__DIR__.'/../config/trustedproxy.php');
 
-        $this->publishes([$source => config_path('trustedproxy.php')]);
+        // ! lumen doenst support publishes
+        if(function_exists('config_path')) {
```
Comment (GrahamCampbell) [inline src/TrustedProxyServiceProvider.php]: cs

File: src/TrustedProxyServiceProvider.php [hunk 2]
```diff
@@ -15,7 +15,10 @@ public function boot()
     {
         $source = realpath(__DIR__.'/../config/trustedproxy.php');
 
-        $this->publishes([$source => config_path('trustedproxy.php')]);
+        // ! lumen doenst support publishes
```
Comment (GrahamCampbell) [inline src/TrustedProxyServiceProvider.php]: "doesn't" is spelt wrong

File: composer.json [hunk 3]
```diff
@@ -10,11 +10,11 @@
         }
     ],
     "require": {
-        "php": ">=5.4.0",
-        "illuminate/contracts": "~5.0"
+        "php": ">=5.5.9",
```
Comment (GrahamCampbell) [inline composer.json]: This doesn't need changing.

File: .travis.yml [hunk 4]
```diff
@@ -1,8 +1,7 @@
 language: php
 
 php:
-  - 5.4
```
Comment (GrahamCampbell) [inline .travis.yml]: php 5.4 doesn't need removing

Comment (GrahamCampbell) [inline .travis.yml]: the tests will still run on 5.0 on php 5.4.

**Comment by vluzrmos** [comment]
Thanks @GrahamCampbell !
All done!

File: composer.json [hunk 5]
```diff
@@ -11,10 +11,10 @@
     ],
     "require": {
         "php": ">=5.4.0",
-        "illuminate/contracts": "~5.0"
+        "illuminate/contracts": "5.0.*|5.1.*"
```
Comment (GrahamCampbell) [inline composer.json]: This change should be totally reverted. `~5.0` already permits 5.1.

File: composer.json [hunk 6]
```diff
@@ -11,10 +11,10 @@
...
     },
     "require-dev": {
-        "illuminate/http": "~5.0",
+        "illuminate/http": "5.0.*|5.1.*",
```
Comment (GrahamCampbell) [inline composer.json]: This change should be totally reverted. `~5.0` already permits 5.1.

**Comment by GrahamCampbell** [comment]
5.1 already worked fine without these changes.

**Comment by GrahamCampbell** [comment]
The only change here is the lumen thing.

**Comment by fideloper** [comment]
(I'll check on this soon, looks great at first glance!)

**Comment by fideloper** [comment]
@GrahamCampbell This look good to you also? Looks good here, just wanting a 2nd pair of eyes :D 
Thanks!

**Comment by vitorbari** [comment]
Just to let you know that i have tested this code and it works great.

**Comment by fideloper** [comment]
Ah, thanks for the reminder

**Pull request merged**